openforcefield/typing/engines/smirnoff/parameters.py

```diff
@@ -1069,6 +1097,33 @@ def __init__(self, **kwargs):
         # TODO: Do we want a docstring here? If not, check that docstring get inherited from ParameterHandler.
         super().__init__(**kwargs)
 
+    def check_handler_compatibility(self,
```

Comment:
I've added an explicit `check_handler_compatibility` function for each ParameterHandler initialized by SMIRNOFF99Frosst. These functions are nearly identical, and could be refactored/consolidated once we get a sense for how well they work.

Reply:
Definitely like the idea of this! And for sure when you get more of a handle on what comparisons can be made there definitely looks to be scope for condensing things, but for now is probably fine (maybe throw in some todo's in case of bus?)!

Reply:
Love it. In general, I think passing around objects (at least in the public API) rather than `dict`s will generally improve encapsulation and thus extensibility.

Reply:
It's probably a good idea to open an issue to remind us to look into code simplification.